scripts/run_project_tests.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+
+"""
+A script to run unity and pytest tests in a host environment for Phoenix-RTOS.
+
+Example of config file:
+
+    campaign_name:
+        unity:
+            - binary1 args
+            - binary2
+        pytest:
+            - path1 args
+            - path2
+"""
+
+import subprocess
+import argparse
+import sys
+import os
+from typing import Dict, List
+
+import yaml
+from colorama import Fore, init
+
+
+class TestSetupError(Exception):
+    pass
+
+
+def run_test(test_cmd: List[str], test_name: str, stream: bool) -> int:
+    """Executes a single test command"""
+    print(f"{Fore.MAGENTA}Running {test_name}: ", flush=True, end="\n" if stream else "")
+
+    if stream:
+        result = subprocess.run(test_cmd, text=True)
+        print(f"{Fore.MAGENTA}Running {test_name}: ", flush=True, end="")
+    else:
+        result = subprocess.run(test_cmd, capture_output=True, text=True)
+
+    if result.returncode != 0:
+        print(f"{Fore.RED}FAIL")
+        if not stream:
+            if result.stdout:
+                print(f"{Fore.YELLOW}STDOUT:")
+                print(result.stdout.strip())
```

Comment:
Not sure why `strip`, if to avoid double newline than this is better (eg. won't strip the indentation at front)
```suggestion
                print(result.stdout, end='')
```